app/src/main/java/com/andrew/tasky/presentation/agenda/AgendaViewModel.kt

```diff
@@ -30,6 +50,35 @@ class AgendaViewModel@Inject constructor(
         }
         .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())
 
+    private fun indexOfTimeNeedle(
+        agendaItems: List<AgendaItem>,
+        currentDateTime: LocalDateTime
+    ): Int {
+        var currentDateTimeIndex = 0
+        for (item in agendaItems) {
+            if (item.startDateAndTime < currentDateTime) {
+                currentDateTimeIndex++
+            }
+        }
+        return currentDateTimeIndex
+    }
+
+    val uiAgendaItems = agendaItems.map { items ->
+        items.map { item ->
+            UiAgendaItem.Item(item)
+        }
+            .toMutableList<UiAgendaItem>()
+            .apply {
+                add(
+                    indexOfTimeNeedle(
+                        agendaItems = items,
+                        currentDateTime = currentDateAndTime.value
+                    ),
+                    UiAgendaItem.TimeNeedle
+                )
+            }
+    }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())
```

Comment:
This is only executed when `agendaItems` changes, but not when `currentDateAndTime` changes. To fix that, you need to use `combine`:
```
val uiAgendaItems = combine(agendaItems, currentDateAndTime) { items, currentTime ->
     items.map { item ->
            UiAgendaItem.Item(item)
        }
            .toMutableList<UiAgendaItem>()
            .apply {
                add(
                    indexOfTimeNeedle(
                        agendaItems = items,
                        currentDateTime = currentTime
                    ),
                    UiAgendaItem.TimeNeedle
                )
            }
}
.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())
```
You also shouldn't need `subscribeToObservables` since `combine` will launch and execute the flow here.